src/main/java/com/example/gradecalculator/config/WebSecurityConfig.java

```diff
@@ -1,26 +1,68 @@
+
 package com.example.gradecalculator.config;
 
+import com.example.gradecalculator.service.UserDetailsServiceImpl;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.web.SecurityFilterChain;
 
+
 @Configuration
 @EnableWebSecurity
 public class WebSecurityConfig {
 
+    private final PasswordConfig passwordConfig;
+
+    @Bean
+    public UserDetailsService userDetailsService() {
+        return new UserDetailsServiceImpl();
+    }
+    public WebSecurityConfig(
+            AuthenticationConfiguration authConfiguration,
+            PasswordConfig passwordConfig) {
+        this.authConfiguration = authConfiguration;
+        this.passwordConfig = passwordConfig;
+    }
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
-        http.authorizeHttpRequests((requests) -> requests.anyRequest().permitAll());
+        http
+                .authorizeRequests((requests) -> requests.anyRequest().permitAll())
+                .httpBasic(Customizer.withDefaults())
+                .formLogin((form) -> form
+                        .loginPage("/login")
+                        .permitAll()
+                )
+                .logout((logout) -> logout
+                        .logoutUrl("/logout")
+                        .logoutSuccessUrl("/index")
+                        .permitAll()
+                );
         return http.build();
     }
-
     @Bean
     public WebSecurityCustomizer webSecurityCustomizer() throws Exception {
         return (web) -> web.ignoring().requestMatchers("/resources/**");
     }
+    private final AuthenticationConfiguration authConfiguration;
```

Comment:
fields should be at the top of the class to improve readablity of the file.

There are multiple conventions that differ from each other, e.g:
https://web.archive.org/web/20130516014426/http://www.oracle.com/technetwork/java/codeconventions-141855.html#1852

You should pick one convention to use in your project and stick to it for all implementations